src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,184 @@
+package chatbot;
+
+import chatbot.exception.InvalidTodoException;
+import chatbot.exception.InvalidEventException;
+import chatbot.exception.InvalidDeadlineException;
+import chatbot.exception.InvalidCommandException;
+import chatbot.exception.InvalidTaskNumberException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Parser {
+    /**
+     * Parses user input for add method. If input is appropriate, returns an array containing the type of task,
+     * description, and other parameters, in that order.
+     *
+     * @param input The command inputted by the user.
+     * @returns An array containing the type and details of the task.
+     * @throws InvalidTodoException If format of todo entered is invalid.
+     * @throws InvalidDeadlineException If format of deadline entered is invalid.
+     * @throws InvalidEventException If format of event entered is invalid.
+     * @throws InvalidCommandException If format of command entered is invalid.
+     */
+    public List<String> parseAdd(String input) throws InvalidTodoException,
```

Comment:
can the name of the method be changed to sound more like a verb? Maybe something like parseAddString